src/transformStateWithClones.js

```diff
@@ -6,8 +6,79 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    currentState = { ...currentState };
+
+    if (action.type === 'addProperties') {
+      Object.assign(currentState, action.extraData);
+      stateHistory.push({ ...currentState });
```

Comment:
[DONT REPEAT YOURSELF] - You are pushing to the stateHistory array after each action. This should be done just once at the end of the loop cycle, outside of the switch or if-else block.